asttokens/mark_tokens.py

```diff
@@ -212,6 +212,12 @@ def visit_call(self, node, first_token, last_token):
     call_start = self._code.find_token(first_child.last_token, token.OP, '(')
     if call_start.index > last_token.index:
       last_token = call_start
+
+    # Handling a python bug with decorators with empty parens, e.g.
+    # @deco()
+    # def ...
+    if util.match_token(first_token, token.OP, '@'):
```

Comment:
Really, the AST considers the start of the call node to be the "@" token? That's disappointing..

Reply:
Made a bug: https://bugs.python.org/issue38535